src/components/cards/TaskCard.tsx

```diff
@@ -47,34 +48,45 @@ export const TaskCard = ({ task, href }: TaskCardProps) => {
     <TaskCardContainer>
       <Stack rowGap={1}>
         <Stack direction="row" justifyContent="space-between">
-          {currentAssignee ? (
-            <Stack direction="row" alignItems="center" columnGap={1}>
-              <CopilotAvatar currentAssignee={currentAssignee as IAssigneeCombined} />
-              <Typography
-                variant="sm"
-                fontSize="12px"
-                sx={{
-                  color: (theme) => theme.color.gray[500],
-                  width: '146px',
-                  overflow: 'hidden',
-                  textOverflow: 'ellipsis',
-                  whiteSpace: 'nowrap',
-                }}
-              >
-                {(currentAssignee as IAssigneeCombined).name === 'No assignee'
-                  ? 'No assignee'
-                  : (currentAssignee as IAssigneeCombined)?.name ||
-                    `${(currentAssignee as IAssigneeCombined)?.givenName ?? ''} ${(currentAssignee as IAssigneeCombined)?.familyName ?? ''}`.trim()}
-              </Typography>
-            </Stack>
-          ) : (
-            <Box>
-              <Stack direction="row" justifyContent="space-between" alignItems="center" columnGap={'4px'}>
-                <Skeleton variant="circular" width={20} height={20} />
-                <Skeleton variant="rectangular" width="146px" height="12px" />
+          <Stack direction="column" rowGap={'4px'}>
+            {task.isArchived && (
+              <Stack direction="row" alignItems={'center'} columnGap={'4px'}>
+                <ArchiveBoxIcon />
+                <Typography variant="bodyXs" sx={{ color: (theme) => theme.color.text.textSecondary }}>
+                  Archived
+                </Typography>
               </Stack>
-            </Box>
-          )}
+            )}
+            {currentAssignee ? (
+              <Stack direction="row" alignItems="center" columnGap={1}>
+                <CopilotAvatar currentAssignee={currentAssignee as IAssigneeCombined} />
+                <Typography
+                  variant="sm"
+                  fontSize="12px"
+                  sx={{
+                    color: (theme) => theme.color.gray[500],
+                    width: '146px',
+                    overflow: 'hidden',
+                    textOverflow: 'ellipsis',
+                    whiteSpace: 'nowrap',
+                  }}
+                >
+                  {(currentAssignee as IAssigneeCombined).name === 'No assignee'
+                    ? 'No assignee'
+                    : (currentAssignee as IAssigneeCombined)?.name ||
+                      `${(currentAssignee as IAssigneeCombined)?.givenName ?? ''} ${(currentAssignee as IAssigneeCombined)?.familyName ?? ''}`.trim()}
```

Comment:
Avoid doing type casting so often. Maybe we can cast this in one place, or have a validation in place. The source of `currentAssignee` should be providing us with the type in the first place.